### ISSUE TITLE
Fix Request: pmd7 AvoidUnnecessaryStringBuilderCreation false positive with while loop

### ISSUE THREAD

**Issue by stokpop** [jborgers/PMD-jPinpoint-rules#413]
```java
    String foo(String value, int len) {
        StringBuilder zeroes = new StringBuilder();

        while (zeroes.length() + value.length() < len) {
            zeroes.append('0');
        }

        return zeroes.append(value).toString(); // <-- false positive, zeroes used in while loop
    }
```

**Comment by stokpop** [comment]
TODO: also check for 
* do/while
* for each
* ?
